### PR TITLE
Use jinja templating to refactor cloudwatch config to make is easier to add and remove an OS

### PR DESCRIPTION
**Pull request by hgreebe**
### Description of changes
* Uses jinja templating to refactor cloudwatch config to make is easier to add and remove an OS.
* Creates a list of default platforms that can be used in the platforms section of a log_config.

### Tests
* Tested locally by running write_cloudwatch_agent.json

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cookbooks/aws-parallelcluster-environment/files/cloudwatch/cloudwatch_agent_config_util.py [hunk 1]
```diff
@@ -59,7 +70,10 @@ def get_input_json(args):
 def _read_json_at(path):
     """Read the JSON file at path."""
     try:
-        with open(path, encoding="utf-8") as input_file:
+        config_args = {
```
Comment (himani2411) [inline cookbooks/aws-parallelcluster-environment/files/cloudwatch/cloudwatch_agent_config_util.py]: Nit: Adding this as a constant would add better readability

File: cookbooks/aws-parallelcluster-environment/files/cloudwatch/write_cloudwatch_agent_json.py [hunk 2]
```diff
@@ -61,9 +64,21 @@ def add_instance_log_stream_prefixes(configs):
     return configs
 
 
+def render_jinja_template(template_file_path, **kwargs):
```
Comment (himani2411) [inline cookbooks/aws-parallelcluster-environment/files/cloudwatch/write_cloudwatch_agent_json.py]: Adding this as part of another python module like `cloudwatch_agent_common_utils.py` whihc can be used by both the other modules would be more beneficial.

So is addition of the `default_platforms` in one single Place as we would need to change this again in two places.

File: cookbooks/aws-parallelcluster-environment/files/cloudwatch/cloudwatch_agent_config_util.py [hunk 3]
```diff
@@ -68,14 +69,26 @@ def _read_json_at(path):
     return None
 
 
+def _read_jinja_template_at(path):
+    """Read the JSON file at path."""
```
Comment (gmarciani) [inline cookbooks/aws-parallelcluster-environment/files/cloudwatch/cloudwatch_agent_config_util.py:73]: Minor: "Read the JSON file at path." => "Read the JSON file at path as a Jinja template"

File: cookbooks/aws-parallelcluster-environment/files/cloudwatch/cloudwatch_agent_config_util.py [hunk 4]
```diff
@@ -68,14 +69,26 @@ def _read_json_at(path):
...
+
+
 def _read_schema():
     """Read the schema for the CloudWatch log configs file."""
     return _read_json_at(SCHEMA_PATH)
 
 
 def _read_log_configs():
     """Read the current version of the CloudWatch log configs file, cloudwatch_agent_config.json."""
-    return _read_json_at(LOG_CONFIGS_PATH)
```
Comment (gmarciani) [inline cookbooks/aws-parallelcluster-environment/files/cloudwatch/cloudwatch_agent_config_util.py:78]: if `_read_json_at` is now unused, let's remove the function.

**Review by himani2411 — APPROVED** [review]

File: cookbooks/aws-parallelcluster-environment/files/cloudwatch/write_cloudwatch_agent_json.py [hunk 5]
```diff
@@ -63,7 +65,7 @@ def add_instance_log_stream_prefixes(configs):
 
 def read_data(config_path):
     """Read in log configuration data from config_path."""
-    with open(config_path, encoding="utf-8") as infile:
+    with open(render_jinja_template(config_path), encoding="utf-8") as infile:
```
Comment (gmarciani) [inline cookbooks/aws-parallelcluster-environment/files/cloudwatch/write_cloudwatch_agent_json.py:68]: `read_data` is a function that given a path returns its content as a JSON dictionary. According to its name and description it's not expected for this function to write a file (done by `render_jinja_template`). My suggestion is to do one of the following:
1. align the name and comment to what the function actually does
2. decouple the generation of the template from the writing of the final file
3. decouple the writing of the final config file

File: cookbooks/aws-parallelcluster-environment/resources/cloudwatch/partial/_cloudwatch_common.rb [hunk 6]
```diff
@@ -129,6 +129,17 @@ def package_path
...
 
+  common_module_path = '/usr/local/bin/cloudwatch_agent_common_utils.py'
+  cookbook_file 'cloudwatch_agent_common_utils.py' do
+    action :create_if_missing
+    source 'cloudwatch/cloudwatch_agent_common_utils.py'
+    cookbook 'aws-parallelcluster-environment'
+    path common_module_path
+    user 'root'
+    group 'root'
+    mode '0644'
```
Comment (gmarciani) [inline cookbooks/aws-parallelcluster-environment/resources/cloudwatch/partial/_cloudwatch_common.rb:140]: This file should be executable by root, not only read/writable.

File: cookbooks/aws-parallelcluster-environment/files/cloudwatch/cloudwatch_agent_config_util.py (hunk 4 above)
Comment (hgreebe) [reply cookbooks/aws-parallelcluster-environment/files/cloudwatch/cloudwatch_agent_config_util.py:78]: It is still being used by _read_schema()

**Pull request merged**